src/main/java/mekanism/common/content/gear/IBlastingItem.java

```diff
@@ -0,0 +1,57 @@
+package mekanism.common.content.gear;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import mekanism.common.content.gear.mekatool.ModuleVeinMiningUnit;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+
+public interface IBlastingItem {
+
+    Map<BlockPos, BlockState> getBlastedBlocks(Level world, Player player, ItemStack stack, BlockPos pos, BlockState state);
+
+    static Map<BlockPos, BlockState> findPositions(Level world, BlockPos targetPos, Direction axis, int radius) {
+        if(radius <= 0) {
+            return Collections.emptyMap();
+        }
+        int minX = 0, minY = 0, minZ = 0;
+        int maxX = 0, maxY = 0, maxZ = 0;
+        switch (axis) {
+            case UP, DOWN -> {
+                minX = -radius;
+                maxX = radius;
+                minZ = -radius;
+                maxZ = radius;
+            }
+            case EAST, WEST -> {
+                minY = -1;
+                maxY = 2 * radius - 1;
```

Comment:
I'm a little confused. This is to make tunneling easier. This makes the bottom of the area you are mining be at your feet so that you can look straight forward while mining horizontally. When looking vertically it will center the area at your target.

Reply:
Ah wait I may just be dumb, I haven't tested it yet due to some irl stuff going on and for some reason felt like the -1 was kind of forgetting how tunneling would work and how an offset of -1 would interact.

Reply:
No worries, for clarification this is the same as how other mod's area mining centers itself. Which doesn't mean we can't do it different, but I do think it is way more convenient than the straightforward approach. I haven't convinced myself that it would be controlled and predictable enough trying to guess based on the angle, but I kinda like the idea so I might think about it some more.

Reply:
The way it works sounds relatively reasonable, the only thing that I wonder if using an angle would be more beneficial for is  cases like this
![2022-05-23_12 39 33](https://user-images.githubusercontent.com/1203288/169876599-458dedeb-7e59-458b-ba55-1f5c3d50e2c2.png) where the voxel shape isn't a basic cube so if for example you are over a "top" face then it will do the going down mining when likely the user wants it to be vertical but creating a step going down one.

Reply:
So I agree that it would be more intuitive to mine vertically in these cases, however I don't think the angle would be a good way to do this as it would be identical to the angle of mining down on the grass beneath it. Because of how it is likely using the face normal to determine the direction I don't know that there is a good way to do this with the current raycast. I don't really like the idea of checking to see if the hit was inside the block and then doing a second raycast against a solid block though to find the actual direction, but it should be possible if we wanted to try that.

Reply:
I wonder if using `Shapes.block()#clip` would work for our uses here?

Reply:
If we switch it to destroyProgress we might need to do that, but I don't think it's firing any events with its current implementation. And if we stick to just doing the player destroy check in the util method I think we're fine, although that might call the event and call getBlastedBlocks unnecessarily.